Maestro hook to update Dockerfiles to pickup latest .NET bits is failing

https://devdiv.visualstudio.com/DevDiv/_build?tracking_data=eyJTb3VyY2UiOiJFbWFpbCIsIlR5cGUiOiJOb3RpZmljYXRpb24iLCJTSUQiOiI1OTQzOCIsIlNUeXBlIjoiUEVSIiwiUmVjaXAiOjEsIl94Y2kiOnsiTklEIjoxNzIzNTIzMiwiTVJlY2lwIjoibTA9MSAiLCJBY3QiOiI3Yjk4MWNhMC1kNWQ3LTQ2YTQtYjU0Zi02ZjFkYTQxOWYzZGYiLCJmZiI6IkNEIn19&buildId=1269882

2018-01-07T20:35:28.2627640Z Invoking app: dotnet E:\A\_work\18\s\update-dependencies\bin\update-dependencies.dll GITHUB_USER=dotnet-maestro-bot GITHUB_EMAIL=dotnet-maestro-bot@microsoft.com GITHUB_PASSWORD=******** CLI_BRANCH=master CLI_RELEASE_PREFIX=2.1.0 RUNTIME_RELEASE_PREFIX=2.2.0
2018-01-07T20:35:28.6692782Z update-dependencies Information: 0 : Looking for the Shared Framework CLI '2.2.0-preview1-007877' depends on.
2018-01-07T20:35:28.6732851Z update-dependencies Information: 0 : Downloading `https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.2.0-preview1-007877/dotnet-sdk-2.2.0-preview1-007877-win-x64.zip`
2018-01-07T20:35:37.8009819Z update-dependencies Information: 0 : Found commit hash 'ac06ffcf2d1bda8d60d6e118e4576665a28a02cb' in `.versions`.
2018-01-07T20:35:37.8039819Z update-dependencies Information: 0 : Downloading `https://raw.githubusercontent.com/dotnet-bot/cli/ac06ffcf2d1bda8d60d6e118e4576665a28a02cb/build/DependencyVersions.props`
2018-01-07T20:35:38.3083339Z ##[error]Failed to update dependencies:
System.InvalidOperationException: Can't find CLI_SharedFrameworkVersion in DependencyVersions.props.
   at Dotnet.Docker.Nightly.CliDependencyHelper.GetSharedFrameworkVersion() in E:\A\_work\18\s\update-dependencies\CliDependencyHelper.cs:line 45
   at Dotnet.Docker.Nightly.Program.UpdateFiles() in E:\A\_work\18\s\update-dependencies\Program.cs:line 83
   at Dotnet.Docker.Nightly.Program.Main(String[] args) in E:\A\_work\18\s\update-dependencies\Program.cs:line 33

This issue appears to be caused by a change to the DepependencyVersions.props file - https://github.com/dotnet/cli/commit/cd86a4cac494d81a68be24c29b9d80b1177ea875#diff-c1e21a47bcc98fb7d424e604f2bbaba3

--- a/update-dependencies/CliDependencyHelper.cs
+++ b/update-dependencies/CliDependencyHelper.cs
@@ -38,11 +38,11 @@ namespace Dotnet.Docker.Nightly
             XNamespace msbuildNamespace = depVersions.Document.Root.GetDefaultNamespace();
             string sharedFrameworkVersion = depVersions.Document.Root
                 .Element(msbuildNamespace + "PropertyGroup")
-                ?.Element(msbuildNamespace + "CLI_SharedFrameworkVersion")
+                ?.Element(msbuildNamespace + "MicrosoftNETCoreAppPackageVersion")
                 ?.Value;
             if (sharedFrameworkVersion == null)
             {
-                throw new InvalidOperationException("Can't find CLI_SharedFrameworkVersion in DependencyVersions.props.");
+                throw new InvalidOperationException("Can't find Shared Framework version in DependencyVersions.props.");
             }
 
             Trace.TraceInformation($"Detected Shared Framework version '{sharedFrameworkVersion}'.");

--- a/update-dependencies/CliDependencyHelper.cs
+++ b/update-dependencies/CliDependencyHelper.cs
@@ -36,13 +36,14 @@ namespace Dotnet.Docker.Nightly
             string cliCommitHash = GetCommitHash();
             XDocument depVersions = DownloadDependencyVersions(cliCommitHash).Result;
             XNamespace msbuildNamespace = depVersions.Document.Root.GetDefaultNamespace();
+            const string sharedFrameworkProperty = "MicrosoftNETCoreAppPackageVersion";
             string sharedFrameworkVersion = depVersions.Document.Root
                 .Element(msbuildNamespace + "PropertyGroup")
-                ?.Element(msbuildNamespace + "MicrosoftNETCoreAppPackageVersion")
+                ?.Element(msbuildNamespace + sharedFrameworkProperty)
                 ?.Value;
             if (sharedFrameworkVersion == null)
             {
-                throw new InvalidOperationException("Can't find Shared Framework version in DependencyVersions.props.");
+                throw new InvalidOperationException($"Can't find '{sharedFrameworkProperty}' in DependencyVersions.props.");
             }
 
             Trace.TraceInformation($"Detected Shared Framework version '{sharedFrameworkVersion}'.");

--- a/update-dependencies/DockerfileShaUpdater.cs
+++ b/update-dependencies/DockerfileShaUpdater.cs
@@ -55,7 +55,8 @@ namespace Dotnet.Docker.Nightly
                     string shaExt = versionEnvName.Contains("SDK") ? ".sha" : ".sha512";
                     string shaUrl = shaMatch.Groups["value"].Value
                         .Replace("dotnetcli", "dotnetclichecksums")
-                        .Replace($"${versionEnvName}", version)
+                        .Replace($"${versionEnvName}", version)     // *nix ENV var reference format
+                        .Replace($"$Env:{versionEnvName}", version) // Windows ENV var reference format
                         + shaExt;
 
                     Trace.TraceInformation($"Downloading '{shaUrl}'.");

